src/encoded/commands/create_annotated_filenames.py

```diff
@@ -654,11 +650,29 @@ def get_aliquot_id_from_samples(tissue_samples: List[Dict[str], Any]) -> Filenam
     aliquot_ids = [
         get_aliquot_id_from_tissue_sample(sample) for sample in tissue_samples
     ]
+    if len(aliquot_ids) > 1:
+        aliquot_ids = get_multiple_aliquot_id_from_samples(aliquot_ids)
     return get_filename_part_for_values(
         aliquot_ids, "tissue aliquot ID", source_name="sample"
     )
 
 
+def get_multiple_aliquot_id_from_samples(ids: List[str]):
+    """Get filename part for files merged from multiple tissue sample aliquots."""
+    aliquot_ids = []
+    core_ids = []
+    for id in ids:
+        aliquot_id = id[:-2]
+        core_id = id[-2:]
```

Comment:
You have an implicit invariant here that must be true in order for calls like this to succeed - so long as there is external validation elsewhere that ensures this, it's probably fine, but may merit a comment to illustrate what you are extracting.